src/docs/asciidoc/configuration.adoc

```diff
@@ -238,9 +270,200 @@ An example `hazelcast-jet.xml` is shown below :
+      jmx-enabled: true
+      collection-interval-seconds: 5
+      retention-seconds: 5
+      metrics-for-data-structures: true
+----
+
+`development-instance-config.yaml`:
+
+[source,yaml]
+----
+hazelcast-jet:
+  instance:
+    flow-control-period: 100
+    backup-count: 1
+    scale-up-delay-millis: 10000
+    lossless-restart-enabled: false
+----
+
+Composing the above two YAML configuration files needs them to be imported as shown below.
+
+[source,yaml]
+----
+hazelcast-jet:
+  import:
+    - development-metrics-config.yaml
+    - development-instance-config.yaml
+----
+
+NOTE: Use the `<import/>` element on top level of the XML hierarchy.
+
+NOTE: Use the `import` mapping on top level of the YAML hierarchy.
+
+Hazelcast Jet and Hazelcast IMDG share same syntax for declarative
+configuration composition. For more examples on importing resources from
+the classpath and file system or resources with variables in their 
+names, refer to the {hz-refman}#composing-declarative-configuration[relevant section] 
+of Hazelcast IMDG reference manual.
+
+=== Using Variables
+In your Hazelcast Jet declarative configuration, you can use variables 
+to set the values of the elements. This is valid when you set a system 
+property programmatically or you use the command line interface. You can
+use a variable in the declarative configuration to access the values of
+the system properties you set.
+
+For example, see the following command that sets two system properties.
+
+```
+-Dcooperative.thread.count=16
+```
+
+Let's get the values of these system properties in the declarative 
+configuration of Hazelcast Jet, as shown below.
+
+In the XML configuration:
+
+[source,xml]
+----
+<hazelcast-jet>
+    <instance>
+       <cooperative-thread-count>${cooperative.thread.count}</cooperative-thread-count>
+    </instance>
+</hazelcast-jet>
+----
+
+In the YAML configuration:
+
+[source,yaml]
+----
+hazelcast-jet:
+  instance:
+    cooperative-thread-count: ${cooperative.thread.count}
+----
+
+If you do not want to rely on the system properties, you can use the 
+`XmlJetConfigBuilder` or `YamlJetConfigBuilder` and explicitly set a 
+`Properties` instance, as shown below.
+
+[source,java]
+----
+Properties properties = new Properties();
+
+// fill the properties, e.g., from database/LDAP, etc.
+
+XmlJetConfigBuilder builder = new XmlJetConfigBuilder();
+builder.setProperties(properties);
+JetConfig config = builder.build();
+JetInstance hz = Jet.newJetInstance(config);
+----
+
+=== Variable Replacers
+
+Hazelcast Jet supports variable replacers which are used to replace 
+custom strings during loading the configuration.
+
+Hazelcast Jet and Hazelcast IMDG share same syntax for variable
+replacers. You can configure your variable replacers in Hazelcast Jet 
+configuration via XML or YAML. Please see {hz-refman}#variable-replacers[Variable Replacers]
+section for more information and examples about variable replacers on 
+Hazelcast IMDG reference manual.
+
```

Comment:
My suggestion for the last sentence:

.... Please see the {hz-refman}#variable-replacers[Variable Replacers section]
in the Hazelcast IMDG Reference Manual for more information and examples on the
variable replacers.